pkg/eventprocessor/impl/executor/impl/executor_test.go

```diff
@@ -0,0 +1,265 @@
+package impl
+
+import (
+	"context"
+	"database/sql"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/stretchr/testify/require"
+	"github.com/textileio/go-tableland/internal/tableland"
+	"github.com/textileio/go-tableland/pkg/eventprocessor"
+	"github.com/textileio/go-tableland/pkg/eventprocessor/eventfeed"
+	"github.com/textileio/go-tableland/pkg/parsing"
+	parserimpl "github.com/textileio/go-tableland/pkg/parsing/impl"
+	"github.com/textileio/go-tableland/pkg/sqlstore/impl/system"
+	"github.com/textileio/go-tableland/pkg/tables/impl/ethereum"
+	"github.com/textileio/go-tableland/tests"
+)
+
+var chainID = 1337
+
+func TestReceiptExists(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	ex, _ := newExecutorWithTable(t, 0)
+
+	txnHash := "0x0000000000000000000000000000000000000000000000000000000000001234"
+
+	bs, err := ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+	ok, err := bs.TxnReceiptExists(ctx, common.HexToHash(txnHash))
+	require.NoError(t, err)
+	require.False(t, ok)
+	require.NoError(t, bs.Commit())
+	require.NoError(t, bs.Close())
+
+	bs, err = ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+	err = bs.SaveTxnReceipts(ctx, []eventprocessor.Receipt{
+		{
+			ChainID:     tableland.ChainID(chainID),
+			BlockNumber: 100,
+			TxnHash:     txnHash,
+		},
+	})
+	require.NoError(t, err)
+	require.NoError(t, bs.Commit())
+	require.NoError(t, bs.Close())
+
+	bs, err = ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+	ok, err = bs.TxnReceiptExists(ctx, common.HexToHash(txnHash))
+	require.NoError(t, err)
+	require.True(t, ok)
+	require.NoError(t, bs.Commit())
+	require.NoError(t, bs.Close())
+
+	require.NoError(t, ex.Close(ctx))
+}
+
+func TestMultiEventTxnBlock(t *testing.T) {
+	t.Parallel()
+
+	// The test setup is executing a block that has the following transactions:
+	// - Txn 1, successful with two events: CREATE a table and INSERT a row.
+	// - Txn 2, failed with two events: CREATE a table and INSERT a row. INSERT has invalid SQL.
+	// - Txn 3, successful with one event: INSERT a row in the table created in Txn 1.
+	//
+	// Checks after block execution:
+	// 1) Txn 1 side-effects are seen in the db.
+	// 2) Txn 2 side-effects aren't seen in the db. In particular, the table created should have been rollbacked.
+	// 3) Txn 3 side-effects are seen in the db. Txn 2 failing should be isolated and have no effect in Txn 3 execution.
```

Comment:
This gives a complete explanation of what is the idea of the tests.
LMK if something isn't clear.

TL;DR: We'll execute 3 transactions in a block. Each of them with two events. Transactions 1 and 3 will succeed. Transaction 2 will fail since the **second** event fails and the first succeeds.

This produces an interesting situation where:
- Txn1 and Txn2 should affect the state of the DB.
- Txn2 should be rollbacked. And extra check that the first event also was rollbacked despite its success.